### DISCUSSION TITLE
物化视图下bitmap的准确性(不是hash碰撞的原因)

### DISCUSSION THREAD

**Discussion by zhangkun-pass** [apache/doris#18678] [A - General / Q&A]
doris版本 1.1.0
bitmap计算的数量实际是2， 走物化视图结果为1

请问这是为什么呢, 从结果上来说这个也不是hash碰撞的原因(之前遇见过, 删除重建物化视图就又正确了) 这是个bug吗

plate_no_hash_id 是通过 bitmap_hash(plate_no) 生成的

sql(本sql不走物化视图 -- 所有存在的字段): 
select DISTINCT plate_no, plate_no_hash_id FROM viid_internal_motorvehicle_default WHERE vehicle_class = "H30"

结果:
![image](https://user-images.githubusercontent.com/28724178/232002244-4ebfdc6e-870f-41bb-b34d-cfa179e5c3f4.png)

##sql  走物化视图的sql(结果为1)
SELECT bitmap_to_string(bitmap_union(TO_BITMAP(plate_no_hash_id))) 
FROM viid_internal_motorvehicle_default WHERE vehicle_class = "H30"

结果
![image](https://user-images.githubusercontent.com/28724178/232002078-3a83a324-de80-4871-a800-d78bd8a2072b.png)

--sql 不走物化视图(结构为2)
SELECT bitmap_to_string(bitmap_union(TO_BITMAP(plate_no_hash_id))) 
FROM viid_internal_motorvehicle_default WHERE vehicle_class = "H30" and source_id = 37

![image](https://user-images.githubusercontent.com/28724178/232002586-06bccb5a-c8f3-477f-8b4c-efb48671588f.png)

**Comment by LemonLiTree** [comment]
可能是已知问题，升级1.1.5看看

**Reply by zhangkun-pass** [reply]
> 可能是已知问题，升级1.1.5看看

您好, 因为并没有在bug记录里没有找到和这个问题相关，  而且这个问题是偶发的  不大好复现，  请问有没有什么手段能够验证哪个版本是没问题的,我好做下版本升级
